### PR TITLE
HTBHF-2741 Add acceptance tests for the address mismatch scenarios.

### PR DESCRIPTION
**Pull request by mikeSmith08**
- Needed to refactor the existing failure scenarios so we could reuse the ClaimFailureScenario enum for these pending decision scenarios.
- The ClaimScenario enum is a rename from ClaimFailureScenario, I'm not sure why its showing as a new class.

### CODE REVIEW

File: src/test/java/uk/gov/dhsc/htbhf/steps/NavigationSteps.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public void navigateToPage(String pageName) {
         page.openDirect();
     }
 
-    @Then("^I am shown the (.*) page")
+    @Then("^I am shown the (.*) page$")
```
Comment (mikeSmith08) [inline src/test/java/uk/gov/dhsc/htbhf/steps/NavigationSteps.java]: I initially added this as the original step matched this with extra words after `page` so the dollar makes it an exact match - it doesn't hurt to keep this in.

File: src/test/resources/features/acceptance/pending-application.feature [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Feature: Pending application
+  In order to apply for the Apply for Healthy Start programme
+  As a potential claimant
+  If my claim is not instantly successful I want feedback about what happens next
+
+  @RequiresWiremock
+  Scenario Outline: An application where the claimant is told we'll let you know
+    Given I am on the first page of the application
+    When I submit an application with details which need reviewing because: <reason>
```
Comment (mikeSmith08) [inline src/test/resources/features/acceptance/pending-application.feature]: I wasn't particularly happy with the naming of this step so happy for any suggestions that come to mind.

File: src/test/resources/features/acceptance/unsuccessful-application.feature [hunk 3]
```diff
@@ -4,14 +4,14 @@ Feature: Unsuccessful application
   If my claim is unsuccessful I want feedback about why my claim was unsuccessful
 
   @RequiresWiremock
-  Scenario Outline: An application where the claimant's identity has not been matched fails
+  Scenario Outline: An instant fail application
```
Comment (mikeSmith08) [inline src/test/resources/features/acceptance/unsuccessful-application.feature]: This was just missed from the previous story.

File: src/test/java/uk/gov/dhsc/htbhf/steps/ClaimScenario.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+@AllArgsConstructor
+@Getter
+public enum ClaimScenario {
+
+    IDENTITY_NOT_MATCHED("claim-identity-failed.json"),
+    ELIGIBILITY_NOT_CONFIRMED("claim-eligibility-failed.json"),
+    DUPLICATE_CLAIM("claim-duplicate.json"),
+    POSTCODE_MISMATCH("claim-postcode-mismatch.json"),
+    FULL_ADDRESS_MISMATCH("claim-full-address-mismatch.json"),
```
Comment (mikeSmith08) [inline src/test/java/uk/gov/dhsc/htbhf/steps/ClaimScenario.java]: Thought this was better than the rather verbose option of `BOTH_ADDRESS_LINE_ONE_AND_POSTCODE_MISMATCH`

File: src/test/java/uk/gov/dhsc/htbhf/steps/DecisionSteps.java [hunk 5]
```diff
@@ -13,10 +13,15 @@
 public class DecisionSteps extends CommonSteps {
 
     @Then("^I am shown an instant failure")
-    public void verifyOnCorrectPage() {
+    public void verifyOnInstantFailurePage() {
         getPages().getUnsuccessfulDecisionPage();
     }
 
+    @Then("^I am shown the decision page saying we'll let you know")
+    public void verifyOnWellLetYouKnowPage() {
```
Comment (YetAnotherMatt) [inline src/test/java/uk/gov/dhsc/htbhf/steps/DecisionSteps.java]: might want to consider renaming this to `verifyOnWeWillLetYouKnowPage`

File: src/test/resources/features/acceptance/pending-application.feature (hunk 2 above)
Comment (YetAnotherMatt) [reply src/test/resources/features/acceptance/pending-application.feature]: `I submit an application that doesn't get an instant answer due to: <reason>`

**Pull request merged**